python/ray/serve/_private/default_impl.py

```diff
@@ -98,20 +117,22 @@ def create_router(
         create_replica_wrapper_func=lambda r: ActorReplicaWrapper(r),
     )
 
-    return Router(
-        controller_handle=controller_handle,
-        deployment_id=deployment_id,
-        handle_id=handle_id,
-        self_actor_id=actor_id,
-        handle_source=handle_options._source,
-        event_loop=event_loop,
-        replica_scheduler=replica_scheduler,
-        # Streaming ObjectRefGenerators are not supported in Ray Client
-        enable_strict_max_ongoing_requests=(
-            not is_inside_ray_client_context
-            and RAY_SERVE_ENABLE_STRICT_MAX_ONGOING_REQUESTS
-        ),
-        resolve_request_args_func=resolve_request_args,
+    return SingletonThreadRouter(
+        AsyncioRouter(
+            controller_handle=controller_handle,
+            deployment_id=deployment_id,
+            handle_id=handle_id,
+            self_actor_id=actor_id,
+            handle_source=handle_options._source,
+            event_loop=event_loop,
```

Comment:
Hmm getting the event loop from a class method on `SingletonThreadRouter`, then feeding it into `AsyncioRouter` for initialization, then feeding that back into `SingletonThreadRouter` seems a bit confusing to me. Would it make more sense for `SingletonThreadRouter` to instantiate `AsyncioRouter` instead of an instance of `AsyncioRouter` being passed in?

Reply:
Yeah I agree that part is a bit funky... the reason I didn't do what you suggested initially was to avoid duplicating all of the init args. But I'm now realizing I could just leave them as passthrough kwargs -- let me do that

Reply:
updated